### PR TITLE
Add evaluation methods to synthesizer

### LINKED ISSUE #1190: Add evaluation methods to synthesizer
### Problem Description

_As a user, it would be useful to evaluate the synthetic data generated against the original data._

### Acceptance criteria

* Add a `evaluation` module and two submodules within it: `single_table` and `multi_table`
* Add the following methods to the `evaluation.single_table` module
    *  `evaluate_quality(real_data, synthetic_data, metadata, verbose)` - Wrapper around the initialization and evaluation of [this class](https://github.com/sdv-dev/SDMetrics/blob/956b9889408c520bebd5e76b026136c7b78225ae/sdmetrics/reports/single_table/quality_report.py#L57).
    * `run_diagnostic(real_data, synthetic_data, metadata, verbose)` - Wrapper around the initialization and evaluation of [this class](https://github.com/sdv-dev/SDMetrics/blob/956b9889408c520bebd5e76b026136c7b78225ae/sdmetrics/reports/single_table/diagnostic_report.py#L70).
    * `get_column_plot(real_data, synthetic_data, metadata, column_name)` - Wrapper around [sdmetrics.reports.utils.get_column_plot](https://github.com/sdv-dev/SDMetrics/blob/29a2ab37e3c81ee4af8604425881afe0fb2cb6ea/sdmetrics/reports/utils.py#L215)
    * `get_column_pair_plot(real_data, synthetic_data, metadata, column_names)` - Wrapper around [sdmetrics.reports.utils.get_column_pair_plot](https://github.com/sdv-dev/SDMetrics/blob/29a2ab37e3c81ee4af8604425881afe0fb2cb6ea/sdmetrics/reports/utils.py#L380)
  
* Add the following methods to the `evaluation.multi_table` module
    * `run_diagnostic(real_data ,synthetic_data, metadata, verbose)` - Wrapper around the initialization and evaluation of [this class](https://github.com/sdv-dev/SDMetrics/blob/956b9889408c520bebd5e76b026136c7b78225ae/sdmetrics/reports/multi_table/diagnostic_report.py#L23).
    * `evaluate_quality(real_data, synthetic_data, metadata, verbose)` - Wrapper around the initialization and evaluation of [this class](https://github.com/sdv-dev/SDMetrics/blob/956b9889408c520bebd5e76b026136c7b78225ae/sdmetrics/reports/multi_table/quality_report.py#L61).
    * `synthesizer.get_column_plot(real_data, synthetic_data, metadata, table_name, column_name)` - Wraps the same method as the single table case but requires the table name.
    * `synthesizer.get_column_pair_plot(real_data, synthetic_data, metadata, table_name, column_names)` - Wraps the same method as the single table case but requires the table name.

### Expected behavior

``` python
# Single table cases

quality_report = evaluate_quality(
  real_data=real_data, # DataFrame
  synthetic_data=synthetic_data, # DataFrame
  metadata=my_metadata, # SingleTableMetadata
  verbose=True
)
diagnostic_report = run_diagnostic(
  real_data=real_data, # DataFrame
  synthetic_data=synthetic_data, # DataFrame
  metadata=my_metadata, # SingleTableMetadata
  verbose=True
)
fig = get_column_plot(
  real_data=real_data, # DataFrame
  synthetic_data=synthetic_data, # DataFrame
  metadata=my_metadata, # SingleTableMetadata
  column_name='age'
)
fig = get_column_plot(
  real_data=real_data, # DataFrame
  synthetic_data=synthetic_data, # DataFrame
  metadata=my_metadata, # SingleTableMetadata
  column_names=['age', 'weight']
)

# Multi-table cases
quality_report = evaluate_quality(
  real_data=real_data, # dictionary
  synthetic_data=synthetic_data, # dictionary
  metadata=my_metadata, # MultiTableMetadata
  verbose=True
)
diagnostic_report = run_diagnostic(
  real_data=real_data, # dictionary
  synthetic_data=synthetic_data, # dictionary
  metadata=my_metadata, # MultiTableMetadata
  verbose=True
)
# Plot the 1D marginal distribution
fig = get_column_plot(
  real_data=real_data, # dictionary
  synthetic_data=synthetic_data, # dictionary
  metadata=my_metadata, # MultiTableMetadta
  table_name='users',
  column_name='age'
)
# Plot the 2D bivariate distribution
fig = get_column_plot(
  real_data=real_data, # dictionary
  synthetic_data=synthetic_data, # dictionary
  metadata=my_metadata, # MultitableMetadata
  table_name='users',
  column_names=['age', 'weight']
)
```

### PR DESCRIPTION
**Pull request by fealho**
Resolve #1190.

### CODE REVIEW

**Review by pvk-developer — APPROVED** [review]
LGTM!

**Review by amontanez24 — APPROVED** [review]
:shipit:

File: sdv/evaluation/multi_table.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            Whether or not to print report summary and progress.
+            Defaults to True.
+
+    Returns:
+        float:
+            The overall quality score.
+    """
+    quality_report = QualityReport()
+    quality_report.generate(real_data, synthetic_data, metadata.to_dict(), verbose)
+    return quality_report.get_score()
```
Comment (npatki) [inline sdv/evaluation/multi_table.py:27]: @fealho this should return the `quality_report` object itself, not the score

File: sdv/evaluation/single_table.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            Whether or not to print report summary and progress.
+            Defaults to True.
+
+    Returns:
+        dict:
+            The diagonstic results.
+    """
+    quality_report = DiagnosticReport()
+    quality_report.generate(real_data, synthetic_data, metadata.to_dict(), verbose)
+    return quality_report.get_results()
```
Comment (npatki) [inline sdv/evaluation/single_table.py:51]: @fealho similarly, this should return the diagnostic_report object itself, not the score

**Pull request merged**